nixos/modules/services/web-apps/discourse.nix

```diff
@@ -555,6 +555,17 @@ in
           + "Either update your PostgreSQL package to the correct version or set services.discourse.database.ignorePostgresqlVersion. "
           + "See https://nixos.org/manual/nixos/stable/index.html#module-postgresql for details on how to upgrade PostgreSQL.";
       }
+      {
+        assertion =
+          databaseActuallyCreateLocally
+          -> (builtins.elem config.services.postgresql.package.pkgs.pgvector (
+            # installedExtensions doesn't exist when when the postgresql package has no extensions
+            config.services.postgresql.package.installedExtensions or [ ]
+          ));
+        message =
+          "Discourse requires the PostgreSQL pgvector extension. Please extend services.postgresql.package accordingly. "
+          + "See https://nixos.org/manual/nixos/stable/index.html#module-postgresql for details.";
+      }
```

Comment:
I don't think this should be necessary if we use the `extensions` option of the `postgresql` module. I have a working version here: https://github.com/talyz/nixpkgs/commit/1bc7000da42bc288e6130530514c76c8409ea0d4. Feel free to pull it in.